tmt/base.py

```diff
@@ -1291,6 +1292,14 @@ def _get_template_content(template: str, template_type: str) -> str:
                 force=force,
                 logger=logger)
 
+            if links.get('verifies') and dry is False:
+                tests = Tree(
+                    path=path,
+                    logger=logger).tests(
+                    names=[directory_path.name],
```

Comment:
I believe this could lead to linking more tests than desired: For example, if there are existing tests with `full` in the name:

    > tmt test ls full
    /tests/execute/upgrade/full
    /tests/init/full
    /tests/pip/install/full

And we create a new test `/tests/run/full`:

    > tmt test create /tests/run/full -t beakerlib --link verifies:...

We would get all matching tests for the linking:

    (Pdb) directory_path.name
    'full'

    (Pdb) [test.name for test in tests]
    ['/tests/execute/upgrade/full', '/tests/init/full', '/tests/pip/install/full', '/tests/run/full']

Why `directory_path.name` is used here instead of the full test `name`?

Reply:
Fixed in e6dbb872.